### DISCUSSION TITLE
Can the memory profiler information be printed in dev console or served in a separate html page?

### DISCUSSION THREAD

**Discussion by jspanchu** [emscripten-core/emscripten#20460] [Ideas]
When the link flag `--memoryprofiler` is used, `library_memoryprofiler.js` inserts itself as a `div` element in the main html. This div does not have an id or class and is very hard to find when the canvas is declared with `position: absolute` or it is embeded in a react or vue component. Is it possible to somehow print the allocation information in the dev console or maybe serve the memory allocation map in a different url for convenience?

**Comment by jspanchu** [comment]
As an example:
![MicrosoftTeams-image](https://github.com/emscripten-core/emscripten/assets/40538987/25660b74-893e-4c97-a463-3dd744e956c9)

I would like to see all this stuff in a different page instead of seeing it below the canvas.

**Comment by sbc100** [comment]
I don't think there has been any work done on the memory profiler in a long time.  I you would like to work on it, or figure out a way to make it extensible perhaps, that would be great!

**Comment by jspanchu** [comment]
Sure I'd love to improve it. Right now, I'm working on other things, but I expect to be back on wasm stuff in a month or two.
